test/src/main_sender.cpp

```diff
@@ -14,131 +14,165 @@
+
+    std::shared_ptr<cadmium::dynamic::modeling::model> generator_con = 
+    cadmium::dynamic::translate::make_dynamic_atomic_model<ApplicationGen, 
+    TIME, const char* >("generator_con" , std::move(i_input_data_control));
+    //updated relative path --Syed Omar
+    string input_data_ack = "test/data/sender_input_test_ack_In.txt";
+    const char * i_input_data_ack = input_data_ack.c_str();
+
+    std::shared_ptr<cadmium::dynamic::modeling::model> generator_ack = 
+    cadmium::dynamic::translate::make_dynamic_atomic_model<ApplicationGen, 
+    TIME, const char* >("generator_ack" , std::move(i_input_data_ack));
+
+
+    /********************************************/
+    /****** SENDER *******************/
+    /********************************************/
+
+    std::shared_ptr<cadmium::dynamic::modeling::model> sender1 = 
+    cadmium::dynamic::translate::make_dynamic_atomic_model<Sender, TIME>("sender1");
 
 
 /************************/
 /*******TOP MODEL********/
 /************************/
-cadmium::dynamic::modeling::Ports iports_TOP = {};
-cadmium::dynamic::modeling::Ports oports_TOP = {typeid(outp_data),typeid(outp_pack),typeid(outp_ack)};
-cadmium::dynamic::modeling::Models submodels_TOP = {generator_con, generator_ack, sender1};
-cadmium::dynamic::modeling::EICs eics_TOP = {};
-cadmium::dynamic::modeling::EOCs eocs_TOP = {
-  cadmium::dynamic::translate::make_EOC<Sender_defs::packetSentOut,outp_pack>("sender1"),
-cadmium::dynamic::translate::make_EOC<Sender_defs::ackReceivedOut,outp_ack>("sender1"),
-cadmium::dynamic::translate::make_EOC<Sender_defs::dataOut,outp_data>("sender1")
-};
-cadmium::dynamic::modeling::ICs ics_TOP = {
-  cadmium::dynamic::translate::make_IC<iestream_input_defs<Message_t>::out,Sender_defs::controlIn>("generator_con","sender1"),
-  cadmium::dynamic::translate::make_IC<iestream_input_defs<Message_t>::out,Sender_defs::ackIn>("generator_ack","sender1")
-};
-std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> TOP = std::make_shared<cadmium::dynamic::modeling::coupled<TIME>>(
- "TOP", 
- submodels_TOP, 
- iports_TOP, 
- oports_TOP, 
- eics_TOP, 
- eocs_TOP, 
- ics_TOP 
-  );
+    cadmium::dynamic::modeling::Ports iports_TOP = {};
+    cadmium::dynamic::modeling::Ports oports_TOP = {
+        typeid(outp_data),typeid(outp_pack),typeid(outp_ack)};
+    cadmium::dynamic::modeling::Models submodels_TOP = {
+        generator_con, generator_ack, sender1};
+    cadmium::dynamic::modeling::EICs eics_TOP = {};
+    cadmium::dynamic::modeling::EOCs eocs_TOP = {
+        cadmium::dynamic::translate::make_EOC<sender_defs::packetSentOut,
+        outp_pack>("sender1"),
+        cadmium::dynamic::translate::make_EOC<sender_defs::ackReceivedOut,
+        outp_ack>("sender1"),
+        cadmium::dynamic::translate::make_EOC<sender_defs::dataOut,
+        outp_data>("sender1")
+    };
+    cadmium::dynamic::modeling::ICs ics_TOP = {
+        cadmium::dynamic::translate::make_IC<iestream_input_defs<message_t>::out,
```

Comment:
Lines are 82 in length. I have done something like this

Reply:
cadmium::dynamic::translate::make_IC
            <iestream_input_defs<Message_t>::out,
                sender_defs::control_in>("generator_con","sender1"),